### PR TITLE
Plot legends expand by default when enabled

### LINKED ISSUE #7403: [Plot Legends] Legends not expanding when requested
#### Summary

This issue is seen in both Stacked Plots and Overlay Plots. However, the behaviors amongst the two are _slightly_ different, but caused by the same issue. 

***_**OVERLAY PLOT ISSUE (Occurs with both Real Telemetry or Sine Wave Generators)**_ ***
When, in an Overlay Plot that is populated either with _real telemetry,_ or _Sine Wave Generators,_ a user selects the "expand by default" option for the legend and saves, the class that enables the legend to be expanded is being applied, but the legend is not actually expanding. Instead, we're getting into a weird state that looks like the image below. Notice that the triangle icon is indeed the one that we use for expanding an item (the triangle is pointing down), but the legend itself is not expanding. For a user to see the expanded legend, they have to manually click the icon to first collapse it, then click it again to expand it. 
Screenshot 2024-01-23 at 1 59 18 PM

Additionally, the inspector settings are not updating once out of edit mode or even refreshed. This image below shows that even after I saved the plot settings with the legend "expanded by default" enabled, then _refreshed_, the inspector still shows the setting as disabled. However, note that if I do go into edit mode and look at the setting, it shows that it is enabled. 
Screenshot 2024-01-23 at 2 00 58 PM

***_**STACKED PLOT ISSUE (Sine Wave Generators and Real Telemetry present different behaviors)**_ ***
(A) When, in a Stacked Plot that is populated with **Sine Wave Generators**, the following incorrect behavior happens when you enable the "expand by default" setting for legends: 
While the inspector shows that the setting is enabled, the legends in the plot are not being expanded on default. Unlike the Overlay Plot issue from above, the "expanded" class is not being added in the code, and thus is not even being shown visually with the triangle icons.  
Screenshot 2024-01-23 at 2 15 37 PM

(B) When, in a Stacked Plot that is populated with **Real Telemetry**, the following incorrect behavior happens when you enable the "expand by default" setting for legends, save the object, and refresh: 
The same behavior as Overlay Plots occurs, where the Inspector still shows the setting as _disabled_, while the legend icon appears to be expanded, while the legend table itself is collapsed. 
Screenshot 2024-01-23 at 2 28 50 PM

#### Expected vs Current Behavior

When the setting for legends, "expand by default" is enabled on a Overlay Plot or Stacked Plot, the legend should be expanded when you first navigate to the object. You should not have to manually expand the legend.

#### Steps to Reproduce

**OVERLAY PLOT ISSUE & STACKED PLOT ISSUE (USING REAL TELEMETRY)_ **
1. Navigate to an Overlay Plot and go into Edit Mode. For Stacked Plots, make sure the plot is populated with REAL telemetry, not Sine Wave Generators.
2. Under the Config tab in the Inspector, enable the "expand by default" option for legends, then Save. 
3. Observe that in the Inspector, under the Config tab, the "expand by default" setting is still set to "No."
4. Also observe that the legend's triangle icon is pointing down (showing the legend is expanded), but the legend table does not appear until you press the triangle icon to collapse, then press again to expand the legend.

**_STACKED PLOT ISSUE (USING SWGs)_ **
1. Navigate to a Stacked Plot that is populated with SWGs. 
2. Under the Config tab in the Inspector, enable the "expand by default" option for legends, then Save. 
3. Observe that while the inspector is showing the "expand by default" setting set to "Yes", the legend in the plot is completely collapsed.

#### Environment

* Open MCT Version:  3.3.0
* Deployment Type:  testathon
* OS:
* Browser: Chrome

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7403 

### Describe your changes:

Rely on Vue to enable/disable legend. Ensure we check config on mount for expansion and properties.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -493,7 +493,7 @@
     "WCAG",
     "stackedplot",
     "Andale",
-    "unnnormalized",
+    "unnormalized",
```
Comment (scottbell) [inline .cspell.json:496]: Drive by to fix misspelling

File: src/plugins/persistence/couch/plugin.js
Comment (scottbell) [inline src/plugins/persistence/couch/plugin.js:1]: All misspelling fixes

File: src/plugins/plot/configuration/XAxisModel.js [hunk 2]
```diff
@@ -94,7 +94,7 @@ export default class XAxisModel extends Model {
    */
   defaultModel(options) {
     const bounds = options.openmct.time.bounds();
-    const timeSystem = options.openmct.time.timeSystem();
+    const timeSystem = options.openmct.time.getTimeSystem();
```
Comment (scottbell) [inline src/plugins/plot/configuration/XAxisModel.js:97]: this warning was popping up that we're violating our own deprecated code!

File: src/plugins/plot/inspector/PlotOptionsItem.vue [hunk 3]
```diff
@@ -132,7 +132,8 @@ export default {
   },
   data() {
     return {
-      expanded: false
+      expanded: false,
+      status: null
```
Comment (scottbell) [inline src/plugins/plot/inspector/PlotOptionsItem.vue:136]: we reference `status` as part of the render, causing vue to puke a warning

File: src/plugins/plot/legend/PlotLegend.vue [hunk 4]
```diff
@@ -35,7 +35,12 @@
 
     <div class="c-plot-legend__wrapper" :class="{ 'is-cursor-locked': cursorLocked }">
       <!-- COLLAPSED PLOT LEGEND -->
-      <div class="plot-wrapper-collapsed-legend" :class="{ 'is-cursor-locked': cursorLocked }">
+      <div
+        v-if="!isLegendExpanded"
```
Comment (scottbell) [inline src/plugins/plot/legend/PlotLegend.vue:39]: add a v-if here instead of just relying on CSS

File: src/plugins/plot/legend/PlotLegend.vue [hunk 5]
```diff
@@ -150,6 +160,7 @@ export default {
   mounted() {
     this.loaded = true;
     this.isLegendExpanded = this.legend.get('expanded') === true;
+    this.$emit('expanded', this.isLegendExpanded);
```
Comment (scottbell) [inline src/plugins/plot/legend/PlotLegend.vue:163]: this causes the `PlotView.vue` to add the appropriate CSS class to the legend

File: src/styles/_legacy-plots.scss [hunk 6]
```diff
@@ -658,13 +658,6 @@ mct-plot {
 
 .gl-plot,
 .c-plot {
-  &.plot-legend-collapsed .plot-wrapper-expanded-legend {
```
Comment (scottbell) [inline src/styles/_legacy-plots.scss:661]: I don't think we need to use these and can instead rely on Vue

File: src/plugins/plot/pluginSpec.js [hunk 7]
```diff
@@ -401,6 +401,7 @@ describe('the plugin', function () {
       const clickEvent = createMouseEvent('click');
 
       legendControl.dispatchEvent(clickEvent);
+      await nextTick();
```
Comment (scottbell) [inline src/plugins/plot/pluginSpec.js:404]: need to wait for next tick as vue is manipulating DOM

File: src/plugins/plot/stackedPlot/pluginSpec.js [hunk 8]
```diff
@@ -360,14 +360,16 @@ describe('the plugin', function () {
...
-    it('Renders an expanded legend for every telemetry', () => {
+    it('Renders an expanded legend for every telemetry', async () => {
       let legendControl = element.querySelector(
         '.c-plot-legend__view-control.gl-plot-legend__view-control.c-disclosure-triangle'
       );
       const clickEvent = createMouseEvent('click');
 
       legendControl.dispatchEvent(clickEvent);
 
+      await nextTick();
```
Comment (scottbell) [inline src/plugins/plot/stackedPlot/pluginSpec.js:371]: need to wait for next tick as vue is manipulating DOM

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 9]
```diff
@@ -63,6 +63,60 @@ test.describe('Overlay Plot', () => {
...
+
+    await page.goto(overlayPlot.url);
+
+    await page.getByRole('tab', { name: 'Config' }).click();
+
+    // Assert that the legend is collapsed by default
+    await expect(page.getByLabel('Plot Legend Collapsed')).toBeVisible();
+    await expect(page.getByLabel('Plot Legend Expanded')).toBeHidden();
+    let expandDefaultValue = await page.getByLabel('Expand by Default').textContent();
+    expect(expandDefaultValue).toBe('No');
```
Comment (ozyx) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:98]: Would this also work?
```suggestion
    await expect(page.getByLabel('Expand by Default')).toHaveText('No');
```

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 10]
```diff
@@ -63,6 +63,60 @@ test.describe('Overlay Plot', () => {
...
+    await page.reload();
+
+    // Assert that the legend is expanded on page load
+    await expect(page.getByLabel('Plot Legend Collapsed')).toBeHidden();
+    await expect(page.getByLabel('Plot Legend Expanded')).toBeVisible();
+    await expect(page.getByRole('cell', { name: 'Name' })).toBeVisible();
+    await expect(page.getByRole('cell', { name: 'Timestamp' })).toBeVisible();
+    await expect(page.getByRole('cell', { name: 'Value' })).toBeVisible();
+    expandDefaultValue = await page.getByLabel('Expand by Default').textContent();
+    expect(expandDefaultValue).toBe('Yes');
```
Comment (ozyx) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:116]: Same thing here
```suggestion
    await expect(page.getByLabel('Expand by Default')).toHaveText('Yes');
```

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 11]
```diff
@@ -63,6 +63,60 @@ test.describe('Overlay Plot', () => {
...
+    expect(await page.getByLabel('Plot Legend Item').count()).toBe(3);
```
Comment (ozyx) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:117]:
```suggestion
    await expect(page.getByLabel('Plot Legend Item')).toHaveCount(3)
```

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 12]
```diff
@@ -63,6 +63,60 @@ test.describe('Overlay Plot', () => {
...
+    expect(expandDefaultValue).toBe('No');
+
+    expect(await page.getByLabel('Plot Legend Item').count()).toBe(3);
+
+    // Change the legend to expand by default
+    await page.getByLabel('Edit Object').click();
+    await page.getByLabel('Expand By Default').check();
+    await page.getByLabel('Save').click();
+    await page.getByRole('listitem', { name: 'Save and Finish Editing' }).click();
+    await page.reload();
```
Comment (ozyx) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:107]: oh man.. these locators are beeeeeaaaautiful ✨

File: src/plugins/plot/configuration/XAxisModel.js (hunk 2 above)
Comment (ozyx) [reply src/plugins/plot/configuration/XAxisModel.js:97]: yeah, we gotta practice what we preach! 😆

File: src/plugins/plot/inspector/PlotOptionsItem.vue (hunk 3 above)
Comment (ozyx) [reply src/plugins/plot/inspector/PlotOptionsItem.vue:136]: nice catch

File: src/plugins/plot/legend/PlotLegendItemCollapsed.vue [hunk 13]
```diff
@@ -22,6 +22,7 @@
 <template>
   <div
     class="plot-legend-item"
+    aria-label="Plot Legend Item"
```
Comment (ozyx) [inline src/plugins/plot/legend/PlotLegendItemCollapsed.vue:25]: should we make these dynamic and include the domain object name?

File: src/plugins/plot/stackedPlot/pluginSpec.js (hunk 8 above)
Comment (ozyx) [reply src/plugins/plot/stackedPlot/pluginSpec.js:371]: as we use Vue more, the more the nextTick() hoops we will have to jump through in the unit tests. As a result I've been removing unit tests that are 100% covered by e2e tests here and there. If this test is stable like this then should be fine to keep it. Hopefully some day we can get actual component testing going here

**Review by ozyx — CHANGES_REQUESTED** [review]
Changes look awesome overall. I have a few comments and suggestions before we merge

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
During testing I noticed that when I toggle the `Expand by Default` option and save it, the change is not immediately applied. A refresh is required for the changes to take effect.

**Comment by scottbell** [comment]
> During testing I noticed that when I toggle the `Expand by Default` option and save it, the change is not immediately applied. A refresh is required for the changes to take effect.

My interpretation of "Expand by Default" was the it should expand the legend on page load, not to immediately expand the legend. Is that what we want?

**Comment by shefalijoshi** [comment]
> > During testing I noticed that when I toggle the `Expand by Default` option and save it, the change is not immediately applied. A refresh is required for the changes to take effect.
> 
> @shefalijoshi My interpretation of "Expand by Default" was the it should expand the legend on page load, not to immediately expand the legend. Is that what we want?

Right. I'm curious about what @charlesh88 thinks. I'll approve this for now. If you feel like the change needs to go into a different PR, I'm ok with that.

**Review by shefalijoshi — APPROVED** [review]

**Comment by scottbell** [comment]
> > > During testing I noticed that when I toggle the `Expand by Default` option and save it, the change is not immediately applied. A refresh is required for the changes to take effect.
> > 
> > 
> > @shefalijoshi My interpretation of "Expand by Default" was the it should expand the legend on page load, not to immediately expand the legend. Is that what we want?
> 
> Right. I'm curious about what @charlesh88 thinks. I'll approve this for now. If you feel like the change needs to go into a different PR, I'm ok with that.

It's been implemented to immediately expand, but can easily be reverted.

**Review by ozyx — APPROVED** [review]

**Pull request merged**